.github/workflows/integration_tests.yml

```diff
@@ -23,28 +29,166 @@ on:
+          platforms: linux/amd64
+          provenance: false
+          push: true
+  
+  deploy_nvcf_test_function:
+    name: Deploy NVCF function
+    runs-on: ubuntu-latest
+    needs: [metadata, build_nvcf_image]
+    if: (inputs.target == 'nvcf' || github.event_name == 'schedule' || inputs.target == 'nightly')
+    environment: backend-validation
+    outputs:
+      nvcf_function_version_id: ${{ steps.deploy.outputs.nvcf_function_version_id }}
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.metadata.outputs.cudaq_commit }}
+          fetch-depth: 1
+
+      - name: Install NGC CLI
+        uses: ./.github/actions/install-ngc-cli
+        with:
+          version: 3.38.0
+          checksum: 427c67684d792b673b63882a6d0cbb8777815095c0f2f31559c1570a91187388
+
+      - name: Deploy NVCF Function
+        id: deploy
+        env:
+          NGC_CLI_API_KEY: ${{ secrets.NGC_NVCF_CREDENTIALS }}
+          NGC_CLI_ORG: ${{ secrets.NGC_NVCF_ORG }}
+          NGC_CLI_TEAM: cuda-quantum
+        run: |
+          create_function_result=$(ngc-cli/ngc cloud-function function create \
+            --container-image nvcr.io/${{ secrets.NGC_NVCF_ORG }}/cuda-quantum/cuda-quantum:nightly \
+            --api-body-format CUSTOM \
+            --inference-port 3030 \
+            --health-uri / \
+            --inference-url /job \
+            --name cudaq-nightly-integration-test \
+            $NVCF_FUNCTION_ID)
+          version_id=$(echo "$create_function_result" | grep 'Version: \S*' |  head -1 | cut -d ':' -f 2 | tr -d ' ')
+          echo "Create version Id: $version_id"
+          echo "nvcf_function_version_id=$version_id" >> $GITHUB_OUTPUT
+          # Deploy it
+          ngc-cli/ngc cloud-function function deploy create --deployment-specification $NGC_NVCF_DEPLOYMENT_SPEC $NVCF_FUNCTION_ID:$version_id
+          function_status=DEPLOYING
+          while [ "$function_status" =  "DEPLOYING" ]; do 
+            echo "Waiting for deploying NVCF function version $version_id ..." 
+            sleep 300 
+            function_info=$(ngc-cli/ngc cloud-function function info $NVCF_FUNCTION_ID:$version_id)
+            function_status=$(echo "$function_info" | grep 'Status: \S*' |  head -1 | cut -d ':' -f 2 | tr -d ' ')
+          done
+          if [ "$function_status" !=  "ACTIVE" ]; then
+            echo "::error:: Failed to deploy NVCF Test Function"
+            exit 1
+          fi
+
+  integration_test:
+    name: Integration test using Docker
+    runs-on: ubuntu-latest
+    needs: [metadata, build_nvcf_image, deploy_nvcf_test_function]
+    # when none of the needed jobs fail or are cancelled (skipped or successful jobs are ok).
+    # For example, if only one hardware backend test is requested (ionq, iqm, etc.), NVCF setup jobs are skipped.
+    if: ${{ always() && !failure() && !cancelled() }} 
```

Comment:
I think it would be cleaner to just create a separate job to validate submission to nvcf. Then you could also make the different tests separate steps to make it easier to find what failed.

Reply:
Agreed. I've refactored nvcf submission into a separate job.